bangazonapi/views/product.py

```diff
@@ -291,3 +294,48 @@ def minprice_filter(product):
         serializer = ProductSerializer(
             products, many=True, context={'request': request})
         return Response(serializer.data)
+
+    @action(methods=['post', 'delete'], detail=True)
+    def like(self, request, pk=None):
+        if request.method == "POST":
+            like = {}
+
+            try:
+                customer = Customer.objects.get(user=request.auth.user)
+                product = Product.objects.get(pk=pk)
+                like['customer'] = customer.id
+                like['product'] = product.id
+                serializer = ProductLikeSerializer(
+                    data=like, context={'request': request})
+                if serializer.is_valid(raise_exception=True):
+                    serializer.save()
+                    return Response(serializer.data)
+            except Product.DoesNotExist as ex:
+                return Response({'message': ex.args[0]}, status=status.HTTP_404_NOT_FOUND)
+            except ValidationError as ex:
+                return Response({"reason": ex.message}, status=status.HTTP_400_BAD_REQUEST)
+        
+        if request.method == "DELETE":
+            customer = Customer.objects.get(user=request.auth.user)
+            try:
+                like = ProductLike.objects.get(customer=customer, product=pk)
+                like.delete()
+                return Response({}, status=status.HTTP_204_NO_CONTENT)
+
+            except ProductLike.DoesNotExist as ex:
+                return Response({'message': ex.args[0]}, status=status.HTTP_404_NOT_FOUND)
+            except Exception as ex:
+                return Response({'message': ex.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+class ProductLikeSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ProductLike
+        validators = [
+            UniqueTogetherValidator(
+                queryset=ProductLike.objects.all(),
+                fields=['customer', 'product'],
+                message="Customer can like a product more than once"
+            )
+        ]
+        fields = ('id', 'customer', 'product')
+        depth = 0
```

Comment:
Depth is set to 0 if it isn't on the serializer so this line doesn't need to be there